### PR TITLE
Bugfix: Create notification when comment added, deleted and thread resolved, deleted

### LINKED ISSUE #6483: [Bug] Notification is not sent when an user is tagged for the first time
When someone is tagged in a comment in a thread for the first time, the tagged user receives an email for the tagging but the user does not receive a notification.

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
When an user mentions another user in the first comment of a thread, the mentioned user does not get a notification but gets the email notification. This PR fixes this issue. It also sends notifications for the following cases:
- A comment is deleted
- Someone is tagged in a comment
- A thread is deleted
- A thread is resolved

Fixes #6483 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Unit tests
- Tested on local machine

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>
No changes to code coverage between the base branch and the head branch</details>

### CODE REVIEW

**Review by sharat87 — COMMENTED** [review]
IIRC, this was done because when a new thread was created, two notifications were being created. One for the comment thread, and another for the comment itself. So we disabled the creation of the notification for the comment, and only have it for the comment thread. Can you confirm if we don't have that problem of duplicate notifications with this PR? (Sorry, I couldn't confirm this by looking at the changes).

Otherwise looks good to me.

**Comment by nayan-rafiq** [comment]
@sharat87 You are right. Earlier we were sending notification to every user in the organization but now we send to current thread users only. So, the notification for the create thread event is not sent to anyone because no one is subscribed to that thread at that time. Thus the notification for the comment was also missing event if someone is tagged in the first comment. Need some idea on this on how to handle the case.

**Review by rishabhsaxena — APPROVED** [review]

File: app/client/src/notifications/NotificationListItem.tsx [hunk 1]
```diff
@@ -123,6 +124,12 @@ function CommentNotification(props: { notification: AppsmithNotification }) {
 
   const _createdAt = createdAt || creationTime;
   const displayName = authorName || authorUsername;
+  let eventName = event;
+  if (!event || event == "CREATED") {
+    eventName = "left";
+  } else if (event == "TAGGED") {
+    eventName = "mentioned you in";
+  }
```
Comment (rishabhsaxena) [inline app/client/src/notifications/NotificationListItem.tsx:132]: nitpick: shall we add expected `event` values here?

File: app/client/src/notifications/NotificationListItem.tsx [hunk 2]
```diff
@@ -215,6 +224,7 @@ function CommentThreadNotification(props: {
 
   const _createdAt = createdAt || creationTime;
   const displayName = authorName || authorUsername;
+  const eventName = event || "updated";
```
Comment (rishabhsaxena) [inline app/client/src/notifications/NotificationListItem.tsx:227]: similar to above: shall we add expected `event` values here?

**Review by sharat87 — APPROVED** [review]

**Pull request merged**